packages/@react-stately/list/src/useListState.ts

```diff
@@ -51,14 +51,23 @@ export function useListState<T extends object>(props: ListProps<T>): ListState<T
 
   let collection = useCollection(props, factory, context, [filter]);
 
+  let [cachedItems, setCachedItems] = useState(items);
+  let cachedCollection = useCollection({...props, items: cachedItems}, factory, context, [filter]);
```

Comment:
instead of caching the entire collection, couldn't we just track when the list has focus and focusedkey changes, grab the next key and store that? if next key ends up not in the collection when we restore focus, that's fine, focus can go to the collection

Reply:
Good idea, refactored to do this, so that the entire collection is no longer cached.